Dfe.PrepareConversions/Dfe.PrepareConversions/Pages/InvoluntaryProject/Summary.cshtml.cs

```diff
@@ -0,0 +1,69 @@
+using Dfe.PrepareConversions.Data.Models;
+using Dfe.PrepareConversions.Data.Models.Establishment;
+using Dfe.PrepareConversions.Data.Models.InvoluntaryProject;
+using Dfe.PrepareConversions.Data.Models.Trust;
+using Dfe.PrepareConversions.Data.Services;
+using Dfe.PrepareConversions.Data.Services.Interfaces;
+using Dfe.PrepareConversions.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Dfe.PrepareConversions.Pages.InvoluntaryProject
+{
+	public class SummaryModel : PageModel
+	{
+		private readonly IGetEstablishment _getEstablishment;
+		private readonly ITrustsRespository _trustRepository;
+		private readonly IAcademyConversionProjectRepository _academyConversionProjectRepository;
+
+		public SummaryModel(IGetEstablishment getEstablishment, ITrustsRespository trustRepository,
+			 IAcademyConversionProjectRepository academyConversionProjectRepository)
+		{
+			_getEstablishment = getEstablishment;
+			_trustRepository = trustRepository;
+			_academyConversionProjectRepository = academyConversionProjectRepository;
+		}
+
+		public EstablishmentResponse Establishment { get; set; }
+		public TrustSummary Trust { get; set; }
+
+
+		public async Task<IActionResult> OnGetAsync(string urn, string ukprn)
+		{
+			Establishment = await _getEstablishment.GetEstablishmentByUrn(urn);
+			Trust = (await _trustRepository.SearchTrusts(ukprn)).Data.First();
```

Comment:
Can this return zero results (causing `.First()` to throw)? How does this approach differ from `GetTrustByUkprn` on line 45?

Reply:
Yes, it would throw. The SearchTrusts endpoint returns a slightly different set of properties.
In the post method I needed to get the GroupId which wasn't available when calling the SearchTrusts endpoint.

I could change line 37 to FirstOrDefault()? Or would you prefere me to update it to call GetTrustByUkprn instead?

Reply:
> The SearchTrusts endpoint returns a slightly different set of properties.

Ah, that's a pain - prob's one for the tech debt backlog.

`FirstOrDefault()` would be a good idea to guard against a potential exception :)

Reply:
Done ✔